src/ui/public/vislib/lib/types/point_series.js

```diff
@@ -130,6 +130,13 @@ export function VislibTypesPointSeries() {
             }
           }
         ];
+      } else {
+        const categoryAxis1 = config.categoryAxes.find((categoryAxis) => {
+          return categoryAxis.id === 'CategoryAxis-1';
+        });
+        if (categoryAxis1) {
+          categoryAxis1.title.text = data.get('xAxisLabel');
```

Comment:
but i am ok with moving forward with this, as this is probably the easiest way to achieve this.
i made a wrong assumption when building this, that axis title should be provided by configuration (and not by data), that title is not a fluid thing that can change, but something fixed ...

obviously that is not the case, if we include dynamic stuff in it (like time intervals)